### PR TITLE
Makes using an upward search operator in an expression work

### LINKED ISSUE #617: Path operators don't work without a left-hand side
image

### PR DESCRIPTION
**Pull request by ike709**
Fixes #617

### CODE REVIEW

File: DMCompiler/DM/Visitors/DMVisitorExpression.cs [hunk 1]
```diff
@@ -47,6 +47,22 @@ internal DMVisitorExpression(DMObject dmObject, DMProc proc, DreamPath? inferred
...
+            {
+                var truePath = new DreamPath(_dmObject.Path + "/" + constant.Value.Path.PathString[1..]);
+                // Check that it exists
+                if (DMObjectTree.TryGetTypeId(truePath, out var _))
+                {
+                    Result = new Expressions.Path(constant.Location, truePath);
+                    return;
+                }
+                // Skip procs
+                if (truePath.FindElement("proc") == -1)
```
Comment (ike709) [inline DMCompiler/DM/Visitors/DMVisitorExpression.cs:61]:
```suggestion
                // Skip procs and verbs
                if (truePath.FindElement("proc") == -1 && truePath.FindElement("verb") == -1)
```

**Review by wixoaGit — CHANGES_REQUESTED** [review]
A better way to handle this would be to add support in the parser for `DMASTUpwardPathSearch` without a path on the left side, implicity using the current path.

Having `DownwardSearch` and `UpwardSearch` path types is actually an incorrect representation of what's going on and I plan to have them removed at some point.

**Comment by ike709** [comment]
Seems to work

**Comment by ike709** [comment]
Nevermind then.

**Comment by ike709** [comment]
Going to just give up on this for now.

**Pull request closed without merging**